packages/core/src/utils.test.ts

```diff
@@ -281,6 +281,17 @@ describe('Core Utils', () => {
   });
 
   test('Deep equals', () => {
+    // Empty values = null, undefined, empty string
+    expect(deepEquals(null, null)).toBe(true);
+    expect(deepEquals(null, undefined)).toBe(true);
+    expect(deepEquals(null, '')).toBe(true);
+    expect(deepEquals(undefined, null)).toBe(true);
+    expect(deepEquals(undefined, undefined)).toBe(true);
+    expect(deepEquals(undefined, '')).toBe(true);
+    expect(deepEquals('', null)).toBe(true);
```

Comment:
This feels a little weird.  Perhaps `deepEquals` should be renamed to something like `fhirEquals` or `deepFhirEquality` to make it clear what's going on.  Thoughts? @rahul1